File: src/main/java/fr/ynerant/leveleditor/api/editor/sprites/Category.java
package fr.ynerant.leveleditor.api.editor.sprites;

import java.util.List;

public class Category {
    private List<Sprite> sprites;
    private String name;

    private Category() {
    }

    public static Category create(String name, List<Sprite> sprites) {
        Category c = new Category();

        c.name = name;
        c.sprites = sprites;

        return c;
    }

    public String getName() {
        return name;
    }

    public List<Sprite> getSprites() {
        return sprites;
    }

}

File: src/main/scala/fr/ynerant/leveleditor/api/editor/sprites/Category.scala
package fr.ynerant.leveleditor.api.editor.sprites

import scala.collection.mutable.ListBuffer

object Category {
	def create(name: String, sprites: ListBuffer[Sprite]): Category = {
		val c = new Category
		c.name = name
		c.sprites = sprites
		c
	}
}

class Category private() {
	private var sprites = null: ListBuffer[Sprite]
	private var name = null: String

	def getName: String = name

	def getSprites: ListBuffer[Sprite] = sprites

	def addSprite(s: Sprite): Unit = {
		this.sprites += s
	}

	override def toString: String = name
}
